src/lightning/data/streaming/writer.py

```diff
@@ -29,11 +29,35 @@
     from torch.utils._pytree import PyTree, tree_flatten, treespec_dumps
 
 
-def _get_data_optimizer_node_rank() -> Optional[int]:
-    node_rank = os.getenv("DATA_OPTIMIZER_NODE_RANK", None)
-    if node_rank is not None:
-        return int(node_rank)
-    return node_rank
+_FORMAT_TO_RATIO = {
+    "kb": 1024,
+    "mb": 1024**2,
+    "gb": 1024**3,
+    "tb": 1024**4,
+    "pb": 1024**5,
+    "eb": 1024**6,
+    "zb": 1024**7,
+    "yb": 1024**8,
+}
+
+
+def _convert_bytes_to_int(bytes_str: str) -> int:
+    """Convert human readable byte format to an integer."""
+    for suffix in _FORMAT_TO_RATIO:
+        bytes_str = bytes_str.lower().strip()
+        if bytes_str.lower().endswith(suffix):
+            try:
+                return int(float(bytes_str[0 : -len(suffix)]) * _FORMAT_TO_RATIO[suffix])
+            except ValueError:
+                raise ValueError(
+                    "".join(
```

Comment:
This is a very complicated way of joining two a muli-line string. This should have simply been

```py
raise ValueError(
    f"Unsupported value/suffix {bytes_str}. Supported suffix are "
    f'{["b"] + list(_FORMAT_TO_RATIO.keys())}.'
)
```